packages/blog/src/api/post/getAlgoliaPosts.ts

```diff
@@ -8,6 +8,8 @@ export type GetAlgoliaPostProps = {
 }
 
 export function getAlgoliaPosts({ query }: GetAlgoliaPostProps) {
-  const index = getAlgoliaClient().initIndex(`${publicRuntimeConfig.ALGOLIA_INDEX_PREFIX}_post`)
+  const indexName = `${publicRuntimeConfig.ALGOLIA_INDEX_PREFIX}_post`
+
+  const index = getAlgoliaClient().initIndex(indexName)
```

Comment:
Do you need to init the algolia index on every search? I have no idea how it works and I might be wrong, but It seems like the index could be initialized just once and moved outside this function. 🤷‍♂️

Reply:
Good catch. Let me try

Reply:
Ok, it is not needed with ONLY one index.

Problem is that algolia is an indexer, so you cannot query AND sort. You have to create sorted indexes so that means if you want to sort by title desc, for example, you have to change the index name.

Supose an scenario where the user can select in a dropdown sort by title ASC and sort by title DESC. And a useState that sets the sorting selection.

This means that the index will be something like `whemotion_ENV_title_sort_asc` or `whemotion_ENV_title_sort_desc` and with that index we have to init algolia with the `init` method.

For the moment there is no sorting, so I will take outside the initialization of Algolia. But in a future if we add sorting we may have to put inside again

Reply:
Ohhhh makes sense! Good to know!